core-curriculum/career-framework/career_framework.json

```diff
@@ -83,12 +83,12 @@
       {
         "name": "java",
         "requiredCurriculum": "required-for-java-engineer",
-        "optionalCurriculum": "optional-for-java-software-engineer"
+        "optionalCurriculum": "optional-for-java-engineer"
```

Comment:
@iPlessmann changed this line (removed software) to align it with the other names.